### PR TITLE
Fix for issue #3783. Bhargava metabolic model returning the correct muscle mass. 

### PR DESCRIPTION
**Pull request by stingjp**
Previously, when adding a muscle to a `Bhargava2004SmoothMuscleMetabolics` component, the muscle mass is calculated based on either the provided muscle mass or specific tension and stored in a member variable via `Bhargava2004SmoothedMuscleMetabolics_MuscleParameters::setMuscleMass()`. [Example here](https://github.com/opensim-org/opensim-core/blob/6e4577d91fffc64d07e14569e7b94667eac68eb6/OpenSim/Simulation/Model/Bhargava2004SmoothedMuscleMetabolics.cpp#L100).

However, during deserialization, setMuscleMass() is never called for all connected muscles leaving the muscle masses set to zero, which obviously produces incorrect metabolic cost values.

Fixes issue [#3783](https://github.com/opensim-org/opensim-core/issues/3783)

### Brief summary of changes
Included the `extendConnectToModel` method in the header file for `Bhargava2004SmoothMuscleMetabolics`, so that it could be called in the .cpp file. In the method, looped through the muscles that are stored in the list property for the metabolics object, and called `setMuscleMass`. 

### Testing I've completed
Tested the changes locally using the edited sandboxSandbox.cpp file. The file now works as a simple test by creating a model, adding the metabolic component, and printing the model. It then loads in the model file, and verifies that the returned muscle mass matches what was originally set. 

### Looking for feedback on...
First pull request - any feedback welcome. 

### CHANGELOG.md (choose one)

- no need to update because this is a patch to existing code, it only fixes a small bug.

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3840)

### CODE REVIEW

File: OpenSim/Sandbox/Moco/sandboxSandbox.cpp [hunk 1]
```diff
@@ -20,10 +20,68 @@
...
+    
+    createHangingMuscleModel();
+    Model met_model = Model("hanging_muscle_met.osim");
+    met_model.initSystem();
+    const auto bharg = met_model.getComponent<Bhargava2004SmoothedMuscleMetabolics>("/metabolic_cost");
+    Bhargava2004SmoothedMuscleMetabolics_MuscleParameters mus = bharg.get_muscle_parameters(0);
+    double testmass = mus.getMuscleMass();
+    double expectedMass = 1.123; // Set the expected mass here
+    
+    if (testmass != expectedMass) {
```
Comment (nickbianco) [inline OpenSim/Sandbox/Moco/sandboxSandbox.cpp:78]: We actually want to move this test to the OpenSim test suite. The sandbox files are just for temporary code snippets and not meant to hold changes that are permanently commited to the repository. 

You should add the test to [testMocoMetabolics.cpp](https://github.com/opensim-org/opensim-core/blob/main/OpenSim/Moco/Test/testMocoMetabolics.cpp). You should create the hanging muscle model, save it to a file, and then commit that file (e.g., testMocoMetabolics_hanging_muscle.osim) to the folder where testMocoMetabolics.cpp lives. Finally, add your test as a new Catch2 test case:

```c++
TEST_CASE("Bhargava2004SmoothedMuscleMetabolics deserialization") {
    Model model = Model("testMocoMetabolics_hanging_muscle.osim");
    model.initSystem();
    const auto metabolics = model.getComponent<Bhargava2004SmoothedMuscleMetabolics>("/metabolic_cost");
    auto muscleParameters = metabolics .get_muscle_parameters(0);
    double mass = muscleParameters .getMuscleMass();
    CHECK(mass == 1.123);    
}
```

Finally, you can undo the changes to sandboxSandbox.cpp.

**Review by nickbianco — CHANGES_REQUESTED** [review]
The fix looks good, but we need to move the test instead the OpenSim test suite. See my comment for instructions.

**Comment by stingjp** [comment]
@nickbianco Thanks for the comments, totally makes sense. I should now have all the changes that you mentioned committed, including adding a short entry to the `CHANGELOG_MOCO.md`. Let me know if anything else is out of place!

**Comment by stingjp** [comment]
Fixed!

File: OpenSim/Moco/Test/testMocoMetabolics.cpp [hunk 2]
```diff
@@ -613,3 +613,12 @@ TEST_CASE("Bhargava2004SmoothedMuscleMetabolics basics") {
...
 }
+
+TEST_CASE("Bhargava2004SmoothedMuscleMetabolics deserialization") {
+    Model model = Model("testMocoMetabolics_hanging_muscle.osim");
+    model.initSystem();
+    const auto metabolics = model.getComponent<Bhargava2004SmoothedMuscleMetabolics>("/metabolic_cost");
+    auto muscleParameters = metabolics.get_muscle_parameters(0);
+    double mass = muscleParameters.getMuscleMass();
+    CHECK(mass == 1.123);    
+}
```
Comment (nickbianco) [inline OpenSim/Moco/Test/testMocoMetabolics.cpp:624]: Add an empty line at the end of this file.

**Review by nickbianco — CHANGES_REQUESTED** [review]
One more small thing to fix I just noticed, then we should be good to go.

File: OpenSim/Moco/Test/testMocoMetabolics.cpp (hunk 2 above)
Comment (stingjp) [reply OpenSim/Moco/Test/testMocoMetabolics.cpp]: Done!

**Comment by stingjp** [comment]
Good catch, done!

**Review by nickbianco — APPROVED** [review]
:lgtm: (looks good to me!)

Now we'll wait for the CI (continuous integration) tests to finish, and if passing, we merge.

**Review by nickbianco — CHANGES_REQUESTED** [review]
@stingjp to fix this failing tests, we need to make the following change in [OpenSim/Moco/Test/CMakeLists.txt](https://github.com/opensim-org/opensim-core/blob/ae433fd21da931994b313a4dadaac856ae9a8794/OpenSim/Moco/Test/CMakeLists.txt#L78):

```c++
MocoAddTest(NAME testMocoMetabolics RESOURCES
                 testMocoMetabolics_hanging_muscle.osim
                 )
```

**Review by nickbianco — APPROVED** [review]
:lgtm:

**Comment by nickbianco** [comment]
Thanks @stingjp!

**Pull request merged**